### PR TITLE
OPSAPS-50504 add more tests, CM accepts only PostgreSQL as ranger DB Type

### PR DESCRIPTION
**Pull request by akanto**

### CODE REVIEW

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/ranger/RangerRoleConfigProvider.java [hunk 1]
```diff
@@ -45,12 +47,17 @@
...
+                if (rangerRdsView.getDatabaseVendor() == DatabaseVendor.POSTGRES) {
+                    variables.add(new ApiClusterTemplateVariable().name("ranger-ranger_database_host").value(rangerRdsView.getHost()));
+                    variables.add(new ApiClusterTemplateVariable().name("ranger-ranger_database_name").value(rangerRdsView.getDatabaseName()));
+                    variables.add(new ApiClusterTemplateVariable().name("ranger-ranger_database_type").value("PostgreSQL"));
+                    variables.add(new ApiClusterTemplateVariable().name("ranger-ranger_database_user").value(rangerRdsView.getConnectionUserName()));
+                    variables.add(new ApiClusterTemplateVariable().name("ranger-ranger_database_password").value(rangerRdsView.getConnectionPassword()));
+                } else {
+                    throw new CloudbreakServiceException("Ranger database supports only PostgreSQL");
+                }
                 break;
```
Comment (smaniraju) [inline template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/ranger/RangerRoleConfigProvider.java]: I think database types supported for Ranger is configurable in SupportedDatabaseProvider.

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/ranger/RangerRoleConfigProvider.java [hunk 2]
```diff
@@ -74,7 +81,16 @@ public boolean isConfigurationNeeded(CmTemplateProcessor cmTemplateProcessor, Te
...
+                filter(rds -> DatabaseType.RANGER.name().equalsIgnoreCase(rds.getType())).collect(Collectors.toList());
+        if (rdsConfigs.size() < 1) {
+            throw new CloudbreakServiceException("Ranger database has not been provided for RANGER_ADMIN component");
+
+        } else if (rdsConfigs.size() > 1) {
+            throw new CloudbreakServiceException("Multiple databases have been provided for RANGER_ADMIN component");
+        }
+
+        return rdsConfigs.get(0);
     }
```
Comment (smaniraju) [inline template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/ranger/RangerRoleConfigProvider.java]: If no database is configured, the inbuilt database is used and only one external database of a particular type can be attached, so this check might not be required.

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/ranger/RangerRoleConfigProvider.java (hunk 1 above)
Comment (akanto) [reply template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/ranger/RangerRoleConfigProvider.java]: Right, so if that code is correct then it shall never reach this point.

 Unfortunately, SupportedDatabaseProvider is combined for Ambari an CM, and CM supports only PostgreSQL

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/ranger/RangerRoleConfigProvider.java (hunk 2 above)
Comment (akanto) [reply template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/ranger/RangerRoleConfigProvider.java]: Inbuild DB, a DB which is launched by CB is also considered as attached DB in this part of the code.

**Pull request merged**